cattle/plugins/volmgr/volmgr.py

```diff
@@ -203,6 +203,20 @@ def update_managed_volume(instance, config, start_config):
     start_config['binds'] = new_binds_map
 
 
+def snapshotExists(snapshot_uuid, volume_uuid):
+    volumes = v.list_volumes(volume_uuid)
+    if volume_uuid not in volumes:
+        return False
+    return snapshot_uuid in volumes[volume_uuid]["Snapshots"]
+
+
+def snapshotExistsInBlockstore(snapshot_uuid, volume_uuid, blockstore_uuid):
+    volumes = v.list_volumes_from_blockstore(volume_uuid, blockstore_uuid)
```

Comment:
I imagine this would be huge?  I don't want to list all snapshots/volumes in S3.  that could be millions :)